plugins/modules/ome_configuration_compliance_baseline.py

```diff
@@ -736,32 +788,66 @@ def validate_remediate_idempotency(module, rest_obj):
     return noncomplaint_devices, baseline_info
 
 
-def create_remediate_payload(noncomplaint_devices, baseline_info, rest_obj):
+def create_remediate_payload(module, noncomplaint_devices, baseline_info, rest_obj):
     ome_version = get_ome_version(rest_obj)
     payload = {
-        "Id": baseline_info["Id"],
+        "Id": baseline_info.get("Id"),
         "Schedule": {
             "RunNow": True,
             "RunLater": False
         }
     }
+    if module.params.get("run_later"):
+        if not validate_cron(module.params.get("cron")):
+            module.exit_json(msg=INVALID_CRON_TIME_MSG, failed=True)
+        validate_time(module, rest_obj)
+        payload['Schedule']['Cron'] = module.params.get("cron")
+        payload['Schedule']['RunNow'] = False
+        payload['Schedule']['RunLater'] = True
+    elif module.params.get("staged_at_reboot"):
+        payload['IsStaged'] = True
+        payload['Schedule'] = {}
     if LooseVersion(ome_version) >= "3.5":
         payload["DeviceIds"] = noncomplaint_devices
     else:
         payload["TargetIds"] = noncomplaint_devices
     return payload
 
 
+def validate_cron(cron_string):
+    cron_pattern = CRON_REGEX
+    return bool(re.match(cron_pattern, cron_string))
+
+
+def validate_time(module, rest_obj):
+    cron_string = module.params.get("cron")
+    try:
+        cron_dt = datetime.datetime.strptime(cron_string, "%S %M %H %d %m ? %Y")
+    except ValueError:
+        module.exit_json(msg=INVALID_CRON_TIME_MSG, failed=True)
+    job_resp = rest_obj.invoke_request('GET', TIME_URI)
+    job_dict = job_resp.json_data
+    time_string = job_dict['SystemTime']
+    provided_time_dt = datetime.datetime.strptime(time_string, '%Y-%m-%d %H:%M:%S.%f')
+    if provided_time_dt > cron_dt:
+        module.exit_json(msg=PAST_TIME_MSG, failed=True)
+
+
 def remediate_baseline(module, rest_obj):
     noncomplaint_devices, baseline_info = validate_remediate_idempotency(module, rest_obj)
-    remediate_payload = create_remediate_payload(noncomplaint_devices, baseline_info, rest_obj)
+    remediate_payload = create_remediate_payload(module, noncomplaint_devices, baseline_info, rest_obj)
     resp = rest_obj.invoke_request('POST', REMEDIATE_BASELINE, data=remediate_payload)
     job_id = resp.json_data
+    if module.params.get("run_later"):
+        schedule_job(module, rest_obj, job_id)
     if module.params.get("job_wait"):
         job_failed, message = rest_obj.job_tracking(job_id, job_wait_sec=module.params["job_wait_timeout"])
         if job_failed is True:
-            module.fail_json(msg=message, job_id=job_id, changed=False)
+            detailed_output = rest_obj.get_job_execution_details(job_id)
+            module.exit_json(msg=message, job_id=job_id, job_details=detailed_output, failed=True)
```

Comment:
Add this key job_details in the above doc "RETURN" section

Reply:
done